builtin/providers/aws/resource_aws_rds_cluster.go

```diff
@@ -499,6 +511,16 @@ func resourceAwsRDSClusterRead(d *schema.ResourceData, meta interface{}) error {
 	d.Set("reader_endpoint", dbc.ReaderEndpoint)
 	d.Set("replication_source_identifier", dbc.ReplicationSourceIdentifier)
 
+	var roles []string
+	for _, r := range dbc.AssociatedRoles {
+		if *r.Status == "ACTIVE" {
+			roles = append(roles, *r.RoleArn)
```

Comment:
I would have `roles` be a map or a set, so the user could query what the `STATUS` is of a role. It looks to me, that if a role is in the `INVALID` state, it's still attached to the db cluster, preventing that role from being attached again.